Add _field_types attribute to GroupedRecord

B0TAxy opened the pull request:
When using `--hash`, `--resolve`, or both, there's a script called *record_modifier.py* that tries to find the fields and then performs the requested action, adding a new field afterward. However, since the `GroupedRecord` class doesn’t have the `self._field_types` attribute, it just crashes every time a `GroupedRecord` is involved. This code fixes that issue.

Fixes #168

B0TAxy commented:
> Thanks! Could you also add a regression test that tests the old broken situation?

Hi,
Thanks for the review. I've added a regression test — please let me know if it's sufficient.

yunzheng reviewed (changes requested):
Looks good, thanks for the pointer where it goes wrong. It seems that dissect.target is using a private attribute to get the fields.

Can you apply the above suggestions by copilot? PR looks good otherwise for merge.

B0TAxy commented:
> Looks good, thanks for the pointer where it goes wrong. It seems that dissect.target is using a private attribute to get the fields.
> 
> Can you apply the above suggestions by copilot? PR looks good otherwise for merge.

I've addressed the suggestions provided by Copilot.
